packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -3434,6 +3360,17 @@ function updateSuspenseListComponent(
     );
     workInProgress.flags |= DidCapture;
   } else {
+    suspenseContext = setDefaultShallowSuspenseListContext(suspenseContext);
+  }
+  pushSuspenseListContext(workInProgress, suspenseContext);
+
+  validateRevealOrder(revealOrder);
+  validateTailOptions(tailMode, revealOrder);
+  validateSuspenseListChildren(newChildren, revealOrder);
+
+  reconcileChildren(current, workInProgress, newChildren, renderLanes);
```

Comment:
`reconcileChildren` can suspend which lead to the `SuspenseListContext` to be unbalanced so it needs to push first.